osu.Framework/Threading/Scheduler.cs

```diff
@@ -200,20 +200,22 @@ public void CancelDelayedTasks()
         /// </summary>
         /// <param name="task">The work to be done.</param>
         /// <param name="forceScheduled">If set to false, the task will be executed immediately if we are on the main thread.</param>
-        /// <returns>Whether we could run without scheduling</returns>
-        public bool Add(Action task, bool forceScheduled = true)
+        /// <returns>The scheduled task, or <c>null</c> if the task was executed immediately.</returns>
+        public ScheduledDelegate Add(Action task, bool forceScheduled = true)
```

Comment:
might be worth putting a `[CanBeNull]` just so we are 101% aware of the nullability.

```suggestion
        [CanBeNull]
        public ScheduledDelegate Add(Action task, bool forceScheduled = true)
```